test/integration/webhook/v1alpha1/workload_test.go

```diff
@@ -70,3 +70,76 @@ var _ = ginkgo.Describe("Workload defaulting webhook", func() {
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Workload validating webhook", func() {
+	ginkgo.Context("When creating a Workload", func() {
+		ginkgo.It("Should validate spec.podSets.count", func() {
+			ginkgo.By("Creating a new Workload")
+			workload := testing.MakeWorkload("workload2", ns.Name).Obj()
+			workload.Spec.PodSets = []v1alpha1.PodSet{
+				{
+					Count: -1,
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "test",
+								Image: "fake-image",
+							},
+						},
+					},
+				},
+			}
+			err := k8sClient.Create(ctx, workload)
+			gomega.Expect(err).Should(gomega.HaveOccurred())
```

Comment:
Verify the error is of the correct type using "k8s.io/apimachinery/pkg/util/errors"